### DISCUSSION TITLE
Deprecation messages are too ephemeral

### DISCUSSION THREAD

**Discussion by danielo515** [LazyVim/LazyVim#2024] [General]
Hello.
It is nice that you take the time to include a deprecation message when you move to a new API or decide to use a new plugin instead of another.
However, deprecation messages are too short lived, and I don't see a clear way to consult them while doing the migration.
In my experience, they only appear once when you are updated your editor, at which time you probably don't want/can leave whatever you are working on to fix your editor, and the next time they will not appear. Will be great if either:

- There were a command to see deprecation messages that have not been yet migrated
- The deprectaion messages included links to documentations where the migration is explained, so you can open them in your browser and do them at a latter time

Regards

**Comment by amaanq** [comment]
`:mess` or `:Noice log`

**Reply by danielo515** [reply]
I assume you posted the answer with your best intent, but I am not that bad of a vim user to not know I can see the messages that appear as popup messages.
However, as I said previously, the deprecation notices only appear on update or installation, and when you restart your editor, they do not appear ever again, so you just need to remember them

**Reply by amaanq** [reply]
Oh I see, then I'm not sure what deprecation messages you mean - I'd expect them to show on restart, sorry for missing that part. It would be nice if you remember it/could mention what it looked like

**Reply by dpetka2001** [reply]
One example I run into, was when `conform` became the default formatter and I was still using `none-ls`. I had a spec for `none-ls` in my config, so the message would appear every time I started Neovim, because I either had to enable the `none-ls` extra or remove the spec from my config and transition to `conform` (which I eventually did). So, not sure what other messages are showing up, but don't persist across restarts. Would be nice if you could provide an example. Same was for `leap` and `flash`. The message would show up every time on restarts, until I either transitioned to flash and removed the leap's spec from my config or I enabled leap extra to keep using it.

**Reply by danielo515** [reply]
That was my exact scenario. I may have fixed it (probably in the wrong way) and that may be the reason why it doesn't appear anymore, I have to check. But still think that having a link to a document where the thing is easier to read, as opposed to a message window within console window will  still be nice. Imagine you complete half of the migration, you will start getting errors, but no longer the helpful message.

**Reply by dpetka2001** [reply]
As long as there's some kind of error in some spec in your config, you will get the error/deprecation message each time you start Neovim until you resolve it. First of all you have to read the message it's showing up either way. Just to show an example, I uncommented the `none-ls` spec I had from before and I get the message
```
`conform.nvim` and `nvim-lint` are now the default formatters and linters in LazyVim.

You can use those plugins together with `none-ls.nvim`,
but you need to enable the `lazyvim.plugins.extras.lsp.none-ls` extra,
for formatting to work correctly.

In case you no longer want to use `none-ls.nvim`, just remove the spec from your config.
```
This shows up no matter how many times I restart Neovim, until I resolve the issue at hand. You can also take a look at [NEWS.md](https://github.com/LazyVim/LazyVim/blob/main/NEWS.md) for recent important changes. This only started with version 10.x release, but I'm pretty sure it will continue going forward. 

Edit: Maybe the `NEWS.md` could be incorporated in the LazyVim website. Feel free to make a request to the maintainer to see if he agrees with something like this.

**Comment by NoneMann** [comment]
how to fix the treesitter, error during compilation
